File: python/7 kyu/399.py
# https://www.codewars.com/kata/6a54d3e91e9fb16ca31cc912/train/python

# Passed

HEX = set("0123456789ABCDEF")

def find_corrupted_byte(dump):
    result = -1
    for idx, possible_hex in enumerate(dump):
        if len(possible_hex) != 2 or any(char not in HEX for char in possible_hex):
            result = idx
            break
            
    return result

output = find_corrupted_byte(["48", "65", "6G", "6C", "6F"])
print(output)

File: nodejs/6 kyu/399.js
// https://www.codewars.com/kata/5274e122fc75c0943d000148/train/javascript

// Passed

function groupByCommas(n) {
  let stringN = String(n);
  let result = "";
  for (let idx = stringN.length - 1; idx >= 0; idx--) {
    if (idx !== stringN.length - 1 && (stringN.length - idx) % 3 === 1) {
      result = "," + result;
    }

    result = stringN[idx] + result;
  }

  return result;
}

const output = groupByCommas(2147483647);
console.log(output);